datafusion/sql/src/query.rs

```diff
@@ -88,10 +90,10 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         let empty_schema = DFSchema::empty();
 
         let skip = skip
-            .map(|o| self.sql_to_expr(o.value, &empty_schema, &mut PlannerContext::new()))
+            .map(|o| self.sql_to_expr(o.value, &empty_schema, planner_context))
```

Comment:
We should use the existing `PlannerContext` that includes `prepare_param_data_types`, instead of creating an empty one.